site/source/tutorials/005_form_widgets/index.md

```diff
@@ -1,26 +1,159 @@
+
+The first thing that we see is a `key` property. As mentioned before, a key is necessary whenever the same type of widget or virtual DOM element will be rendered by a widget. After that comes the `invalid` property. Setting this property to `true` will update the widget's appearance to indicate that the widget contains invalid data. Next comes the the `label`, `placeholder`, and `type` fields which map to the expected properties.
+
+The `value` property renders the value of the field that is passed into the widget via its properties. Notice that there is no code that manipulates this value within the widget. As parts a [reactive framework](https://en.wikipedia.org/wiki/Reactive_programming), Dojo 2 widgets don't normally update their own state. Rather, they inform their parents that a change has occurred via events or some other mechanism. The parent will then pass updated properties back into the widget after all of the changes have been processed. This allows Dojo 2 applications to centralize data and keep the entire application synchronized.
+
+Finally, the `onChange` and `onBlur` properties are populated with [arrow functions](https://www.typescriptlang.org/docs/handbook/functions.html) that call the corresponding `WorkerForm` event handler. This is another common pattern within Dojo 2 applications - the `WorkerForm` doesn't expose any of the components that it is using to build the form. Rather, it manages its children internally and, if necessary, raises its events to inform its parents of any changes they might be interested in. This frees the consumers of the `WorkerForm` widget from having to understand the internal structure of the widget. Additionally, it allows the `WorkerForm` to change its implementation without affecting its parents as long as it continues to honor the `WorkerFormProperties` interface.
+
+The last change that needs to be made in the `WorkerForm` is to update the `_onSubmit` method to delegate to the `onSubmit` property when it is called. Replace that method with this:
+
+{% include_codefile 'demo/finished/biz-e-corp/src/widgets/WorkerForm.ts' lines:26-31 %}
+
+## Integrating the WorkerForm into the App
+
+Now that the `WorkerForm` widget is done, it's time to update the `App` class to take advantage of it. First, we need to address how to store the form's data as the user fills it out. Recall that the `WorkerForm` will accept an `onChange` property that will allow the `App` class to be informed when a field value changes. However, the `App` class doesn't currently have a place to store those changes. We could add private fields in the class to store those values, but that would be difficult to maintain over time as the application gets more complicated and more data needs to be stored. Instead, we are going to take advantage of another mixin that comes with Dojo 2 that is designed to handle state for us - the `StatefulMixin`.
```

Comment:
It might be worthwhile as an aside for Dojo 1.x users to know the differences between Dojo 1 and 2's interpretation of StatefulMixin?